ProcessLib/ThermalTwoPhaseFlowWithPP/ThermalTwoPhaseFlowWithPPMaterialProperties.cpp

```diff
@@ -0,0 +1,338 @@
+    const int /*dim*/) const
+{
+    return _intrinsic_permeability_models[material_id];
+}
+
+double ThermalTwoPhaseFlowWithPPMaterialProperties::getPorosity(const int material_id,
+    const double /*t*/, const ProcessLib::SpatialPosition& /*pos*/,
+    const double /*p*/, const double T, const double porosity_variable) const
+{
+    return
+        _porosity_models[material_id]->getValue(porosity_variable, T);
+}
+
+double
+ThermalTwoPhaseFlowWithPPMaterialProperties::getNonwetRelativePermeability(
+    const double /*t*/, const ProcessLib::SpatialPosition& /*pos*/,
+    const double /*p*/, const double /*T*/, const double saturation) const
+{
+    const double Se = (saturation - 0.15) / (1 - 0.15);
+    if (saturation < 0.15)
+        return 1.0;
+    return pow(1 - Se, 3);
+}
+
+double ThermalTwoPhaseFlowWithPPMaterialProperties::getWetRelativePermeability(
+    const double /*t*/, const ProcessLib::SpatialPosition& /*pos*/,
+    const double /*p*/, const double /*T*/, const double saturation) const
+{
+    const double Se = (saturation - 0.15) / (1 - 0.15);
+    if (Se < 0)
+        return 0.0;
+    return pow(Se, 3);
+}
+
+double ThermalTwoPhaseFlowWithPPMaterialProperties::getSaturation(const int material_id,
+    const double /*t*/, const ProcessLib::SpatialPosition& /*pos*/,
+    const double /*p*/, const double /*T*/, const double pc) const
+{
+    return _capillary_pressure_models[material_id]->getSaturation(pc);
+}
+
+double ThermalTwoPhaseFlowWithPPMaterialProperties::getCapillaryPressure(const int material_id,
+    const double /*t*/, const ProcessLib::SpatialPosition& /*pos*/,
+    const double /*p*/, const double /*T*/, const double saturation) const
+{
+    return _capillary_pressure_models[material_id]
+        ->getCapillaryPressure(saturation);
+}
+
+double ThermalTwoPhaseFlowWithPPMaterialProperties::getSaturationDerivative(const int material_id,
+    const double /*t*/, const ProcessLib::SpatialPosition& /*pos*/,
+    const double /*p*/, const double /*T*/, const double saturation) const
+{
+    const double dpcdsw =
+        _capillary_pressure_models[material_id]->getdPcdS(saturation);
+    const double dswdpc = 1 / dpcdsw;
+    return dswdpc;
+}
+
+double
+ThermalTwoPhaseFlowWithPPMaterialProperties::calculateUnsatHeatConductivity(
+    double const /*t*/, ProcessLib::SpatialPosition const& /*x*/,
+    double const Sw, double const lambda_pm_dry,
+    double const lambda_pm_wet) const
+{
+    double lambda_pm =
+        lambda_pm_dry + std::pow(Sw, 0.5) * (lambda_pm_wet - lambda_pm_dry);
+    if (Sw > 1)
+        lambda_pm = lambda_pm_wet;
+    else if (Sw < 0)
+        lambda_pm = lambda_pm_dry;
+    return lambda_pm;
+}
+double
+ThermalTwoPhaseFlowWithPPMaterialProperties::calculateSaturatedVaporPressure(
+    const double T) const
+{
+    const double T_0 = 373.15;
+    const double p_0 = 101325.0;
+    const double h_wg = 2258000.0;
+    return p_0 * exp(((1 / T_0) - (1 / T)) * Water * h_wg / IdealGasConstant);
```

Comment:
This is for water vapour system. If the vapour model is fixed here, the process name should be changed to reflect this air water vapour system. I suggest to make it as a material model, e.g. Vapour model